lib/executor.js

```diff
@@ -330,24 +332,37 @@ function setupMiddleware(app, instructions) {
     }
     assert(typeof factory === 'function',
       'Middleware factory must be a function');
-    data.config = getUpdatedConfigObject(app, data.config);
+    var useEnvVar = true;
+    data.config = getUpdatedConfigObject(app, data.config, useEnvVar);
     app.middlewareFromConfig(factory, data.config);
   });
 }
 
-function getUpdatedConfigObject(app, config) {
+function getUpdatedConfigObject(app, config, useEnvVar) {
   var DYNAMIC_CONFIG_PARAM = /\$\{(\w+)\}$/;
 
   function getConfigVariable(param) {
     var configVariable = param;
     var match = configVariable.match(DYNAMIC_CONFIG_PARAM);
     if (match) {
-      var appValue = app.get(match[1]);
-      if (appValue !== undefined) {
+      if (useEnvVar && process.env[match[1]] !== undefined) {
```

Comment:
Please save `match[1]` to a named variable (e.g. `varName`) to make the code easier to read.